python/cudf/cudf/core/indexed_frame.py

```diff
@@ -695,6 +701,119 @@ def _align_to_index(
 
         return result
 
+    def round(self, decimals=0, how="half_even"):
+        """
+        Round to a variable number of decimal places.
```

Comment:
Is moving this function `round` from `Frame` to `IndexedFrame` part of the declared aim of this PR (rounding datetimes)? If so, can you explain the conflict? If not, I would do this in a separate PR.

Reply:
Okay, I see why this helps. In an ideal world, the changes to `Index` would be a separate PR that gets merged before this one. If you have time, I would consider doing that. If you don't have time to split the PR, I would edit this PR description to list the reasons you gave above for why this change is being made here. PR descriptions are an excellent place to document _why_ changes are being made, because that motivation is not always clear to reviewers.

Reply:
That's a good point @bdice! I will keep that in mind, thanks!